src/header.rs

```diff
@@ -0,0 +1,243 @@
+        (_, _)                              => BITRATE_INDEX[4][bitrate_index],
+    };
+
+    let sampling_rate_index = ((header[2] & 0b00001100u8) >> 2) as usize;
+
+    if sampling_rate_index == 3 {
+        return Err(Mp3Error::HeaderError);
+    }
+
+    let sampling_rate = match &version {
+        &Version::Mpeg1   => Ok(SAMPLING_RATE[0][sampling_rate_index]),
+        &Version::Mpeg2   => Ok(SAMPLING_RATE[1][sampling_rate_index]),
+        &Version::Mpeg2_5 => Ok(SAMPLING_RATE[2][sampling_rate_index]),
+        _                 => Err(Mp3Error::HeaderError),
+    }?;
+
+    let padding = match header[2] & 0b00000010u8 {
+        0u8          => Padding::No,
+        0b00000010u8 => Padding::Yes,
+        _            => unreachable!(),
+    };
+
+    let private = match header[2] & 1u8 {
+        0u8 => Private::No,
+        1u8 => Private::Yes,
+        _   => unreachable!(),
+    };
+
+    let mode = match header[3] & 0b11000000u8 {
+        0u8        => Mode::Stereo,
+        0b01000000 => Mode::JointStereo,
+        0b10000000 => Mode::DualChannel,
+        0b11000000 => Mode::Mono,
+        _          => unreachable!(),
+    };
+
+    let mode_extension_bits = header[3] & 0b00110000u8;
+
+    let mode_extension = match &layer {
+        &Layer::LayerIII => match mode_extension_bits {
+            0u8          => ModeExtension::Stereo(false, false),
+            0b00010000u8 => ModeExtension::Stereo(true, false),
+            0b00100000u8 => ModeExtension::Stereo(false, true),
+            0b00110000u8 => ModeExtension::Stereo(true, true),
+            _            => unreachable!(),
+        },
+        &Layer::LayerI | &Layer::LayerII => match mode_extension_bits {
+            0u8          => ModeExtension::Bands(4),
+            0b00010000u8 => ModeExtension::Bands(8),
+            0b00100000u8 => ModeExtension::Bands(12),
+            0b00110000u8 => ModeExtension::Bands(16),
+            _            => unreachable!(),
+        },
+    };
+
+    let copyright = match header[3] & 0b00001000u8 {
+        0b00000000u8 => Copyright::No,
+        0b00001000u8 => Copyright::Yes,
+        _            => unreachable!(),
+    };
+
+    let original = match header[3] & 0b00000100u8 {
+        0b00000000u8 => Original::No,
+        0b00000100u8 => Original::Yes,
+        _            => unreachable!(),
+    };
+
+    let emphasis = match header[3] & 0b00000011u8 {
+        0b00000000u8 => Emphasis::None,
+        0b00000001u8 => Emphasis::Ms50_15,
+        0b00000010u8 => Emphasis::Reserved,
+        0b00000011u8 => Emphasis::CcittJ17,
+        _            => unreachable!(),
+    };
+
+    Ok(FrameHeader {
+        version,
+        layer,
+        protection,
+        bitrate,
+        sampling_rate,
+        padding,
+        private,
+        mode,
+        mode_extension,
+        copyright,
+        original,
+        emphasis,
+    })
+}
+
+impl FrameHeader {
+    pub fn bitrate(&self) -> u16 {
```

Comment:
Is it really worth it to have these methods as opposed to making the fields public?